src/components/SeedInfo/SeedInfoViews/FungalShifts.tsx

```diff
@@ -1,83 +1,110 @@
-import React, { useContext } from 'react';
-import classnames from 'classnames';
+import { useContext } from 'react';
 import { Stack, Form, Tooltip, OverlayTrigger } from 'react-bootstrap';
 
 import GameInfoProvider from '../../../services/SeedInfo/infoHandler';
 import { capitalize } from '../../../services/helpers';
 import { FungalInfoProvider } from '../../../services/SeedInfo/infoHandler/InfoProviders/Fungal';
 import { AlchemyConfigContext } from '../../AlchemyConfigContext';
 import { useTranslation } from 'react-i18next';
-import classNames from 'classnames';
 import { useMaterialFavorite } from './helpers';
+import { MaterialInfoProvider } from '../../../services/SeedInfo/infoHandler/InfoProviders/Material';
 
 const Flask = () => {
 	return <b className="text-info">Flask</b>;
 };
 
+interface IFungalMaterialProps {
+	materials: Map<string, string>
+	direction: boolean
+	isFlask: boolean
+	isFavorite // TODO: add type 
+	showId // TODO: add type 
+}
+
+const FungalMaterial = (props: IFungalMaterialProps) => {
+	const key = props.direction ? "to" : "from"
+
+	/*
+	Var 'props.materials' may contain multiple materials with the same display name.
+	This is because some materials like 'Flammable Gas' have static variants that are
+	pre-rendered in a scene. Static materials have their own material ID.
+
+	These materials should not be displayed twice.
+
+	To solve this, materials will be sorted in a new `Map` where the key is the material
+	name acts as the key and the material ids are stored in the value as an array.
+	*/
+
+	const materialsByName = new Map()
+	props.materials.forEach((matName, matId) => {
+		const ids = materialsByName.get(matName) ?? []
+		ids.push(matId)
+		materialsByName.set(matName, ids)
+	});
+
+	// The logic that returns a JSX component requires an `Array`
+	const materialsByNameArray = Array.from(materialsByName)
+
+	return(
+		<Stack>
+			{props.isFlask && <Flask />}
+			{materialsByNameArray.map(([matName, matIds]) => {
+				return(
+					<div key={`${matName}-a`}>
+						<div key={`${matName}-b`} className={matIds.some(props.isFavorite) ? 'text-info' : ''}>
```

Comment:
`key` isn't needed here - it's not a dynamic array

Reply:
I like this! `matIds.some(props.isFavorite)`

Reply:
> `key` isn't needed here - it's not a dynamic array

When I omit `key` the server throws a bunch of warnings.

```
Warning: Each child in a list should have a unique "key" prop.

Check the render method of `FungalMaterial`. See https://reactjs.org/link/warning-keys for more information.
    at div
    at FungalMaterial (http://localhost:3000/static/js/src_components_SeedInfo_SeedDataOutput_tsx.chunk.js:1267:11)
    at td
    at tr
    at Shift (http://localhost:3000/static/js/src_components_SeedInfo_SeedDataOutput_tsx.chunk.js:1311:13)
    at tbody
    at table
    at FungalShifts (http://localhost:3000/static/js/src_components_SeedInfo_SeedDataOutput_tsx.chunk.js:1410:13)
    at WithShow (http://localhost:3000/static/js/src_components_SeedInfo_SeedDataOutput_tsx.chunk.js:576:84)
    at div
    at http://localhost:3000/static/js/vendors-node_modules_react-bootstrap_esm_Form_js-node_modules_react-bootstrap_esm_ListGroup_j-7324a3.chunk.js:1200:8
    at div
    at http://localhost:3000/static/js/vendors-node_modules_react-bootstrap_esm_Form_js-node_modules_react-bootstrap_esm_ListGroup_j-7324a3.chunk.js:2586:3
    at SeedInfo (http://localhost:3000/static/js/src_components_SeedInfo_SeedDataOutput_tsx.chunk.js:590:13)
    at div
    at http://localhost:3000/static/js/bundle.js:5253:7
    at SeedDataOutput (http://localhost:3000/static/js/src_components_SeedInfo_SeedDataOutput_tsx.chunk.js:467:13)
    at div
    at http://localhost:3000/static/js/bundle.js:5253:7
    at div
    at SeedData (http://localhost:3000/static/js/src_components_SeedInfo_RecipesForSeed_tsx.chunk.js:283:110)
    at InnerLoadable (http://localhost:3000/static/js/bundle.js:270:34)
    at LoadableWithChunkExtractor
    at Loadable
    at Suspense
    at LazySeedInfo
    at div
    at Transition (http://localhost:3000/static/js/vendors-node_modules_restart_ui_esm_Anchor_js-node_modules_restart_ui_esm_Nav_js-node_modules-c3468b.chunk.js:3588:30)
    at http://localhost:3000/static/js/vendors-node_modules_restart_ui_esm_Anchor_js-node_modules_restart_ui_esm_Nav_js-node_modules-c3468b.chunk.js:2866:3
    at http://localhost:3000/static/js/vendors-node_modules_restart_ui_esm_Anchor_js-node_modules_restart_ui_esm_Nav_js-node_modules-c3468b.chunk.js:2736:3
    at http://localhost:3000/static/js/vendors-node_modules_react-bootstrap_esm_Tab_js-node_modules_react-bootstrap_esm_Tabs_js.chunk.js:789:3
    at div
    at http://localhost:3000/static/js/vendors-node_modules_restart_ui_esm_Anchor_js-node_modules_restart_ui_esm_Nav_js-node_modules-c3468b.chunk.js:2956:5
    at Tabs (http://localhost:3000/static/js/vendors-node_modules_react-bootstrap_esm_Tab_js-node_modules_react-bootstrap_esm_Tabs_js.chunk.js:299:9)
    at Tabs (http://localhost:3000/static/js/vendors-node_modules_react-bootstrap_esm_Tab_js-node_modules_react-bootstrap_esm_Tabs_js.chunk.js:932:70)
    at div
    at http://localhost:3000/static/js/bundle.js:4991:3
    at Body (http://localhost:3000/static/js/src_components_Body_tsx.chunk.js:42:110)
    at InnerLoadable (http://localhost:3000/static/js/bundle.js:270:34)
    at LoadableWithChunkExtractor
    at Loadable
    at Suspense
    at LazyBody
    at AlchemyConfigProvider (http://localhost:3000/static/js/bundle.js:48905:100)
    at Suspense
    at ThemeProvider (http://localhost:3000/static/js/bundle.js:49867:100)
    at Router (http://localhost:3000/static/js/bundle.js:38407:15)
    at BrowserRouter (http://localhost:3000/static/js/bundle.js:37248:5)
    at div
    at div
    at DBErrorHandler (http://localhost:3000/static/js/bundle.js:49430:88)
    at App (http://localhost:3000/static/js/bundle.js:49452:82)

```

Reply:
Fixed. Thanks for the explanation.